src/test/ui/issue-26638.stderr

```diff
@@ -6,15 +6,6 @@ LL | fn parse_type(iter: Box<Iterator<Item=&str>+'static>) -> &str { iter.next()
    |
    = help: this function's return type contains a borrowed value, but the signature does not say which one of `iter`'s 2 lifetimes it is borrowed from
 
-error[E0106]: missing lifetime specifier
-  --> $DIR/issue-26638.rs:14:40
-   |
-LL | fn parse_type_2(iter: fn(&u8)->&u8) -> &str { iter() }
-   |                                        ^ expected lifetime parameter
-   |
-   = help: this function's return type contains a borrowed value with an elided lifetime, but the lifetime cannot be derived from the arguments
-   = help: consider giving it an explicit bounded or 'static lifetime
-
```

Comment:
@nikomatsakis what's going wrong here? error checking is skipped for this function.